File: Exercise/LuoGu/Code_2023_7_21/P1089.java
package Code_2023_7_21;

import java.util.Scanner;

public class P1089 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = 12;
        int totalSave = 0, save, balance = 0, plan, remainder;
        boolean isRisk = false;
        int minRiskMonth = 0;
        for (int i = 0; i < n; i++) {
            plan = sc.nextInt();
            if (isRisk) {
                continue;
            }
            remainder = balance + 300 - plan;
            if (remainder < 0) {
                isRisk = true;
                minRiskMonth = i + 1;
            }
            save = remainder / 100 * 100;
            totalSave += save;
            balance = remainder - save;
        }

        if (isRisk) {
            System.out.println(-minRiskMonth);
        } else {
            System.out.println((int) (totalSave * 1.2 + balance));
        }

        sc.close();
    }
}

File: Exercise/LuoGu/Code_2023_7_21/P1089.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{

    int n = 12;
    int save = 0, balance = 0, plan, rem;
    bool is_risk = false;
    int min_month;
    for (int i = 0; i < n; i++)
    {
        cin >> plan;
        if (is_risk)
            continue;
        rem = balance + 300 - plan;
        if (rem < 0)
        {
            min_month = i + 1;
            is_risk = true;
        }
        save += rem / 100 * 100;
        balance = rem - rem / 100 * 100;
    }
    if (is_risk)
    {
        cout << -min_month << endl;
    }
    else
    {
        cout << save * 1.2 + balance << endl;
    }

    return 0;
}